core/trino-main/src/main/java/io/trino/execution/buffer/BroadcastOutputBuffer.java

```diff
@@ -370,7 +370,8 @@ private synchronized ClientBuffer getBuffer(OutputBufferId id)
         // without a clean "no-more-buffers" message from the scheduler.  This happens with limit queries and is ok because
         // the buffer will be immediately destroyed.
         BufferState state = stateMachine.getState();
-        checkState(state.canAddBuffers() || !outputBuffers.isNoMoreBufferIds(), "No more buffers already set");
+        // buffer may become aborted while the final output buffers are being set
+        checkState(state == ABORTED || state.canAddBuffers() || !outputBuffers.isNoMoreBufferIds(), "No more buffers already set");
```

Comment:
I don't understand, we already have `if (state != ABORTED) {` check below. It was always `true` before?

Reply:
Not necessarily. 

See `state.canAddBuffers() || !outputBuffers.isNoMoreBufferIds()`

When the buffer state is `ABORTED` the `state.canAddBuffers()` check doesn't pass, however it is possible for the `!outputBuffers.isNoMoreBufferIds()` to pass.